x-pack/plugins/endpoint/common/types.ts

```diff
@@ -23,20 +23,56 @@ export type ImmutableSet<T> = ReadonlySet<Immutable<T>>;
 export type ImmutableObject<T> = { readonly [K in keyof T]: Immutable<T[K]> };
 
 export class EndpointAppConstants {
+  static ALERT_INDEX_NAME = 'my-index';
   static ENDPOINT_INDEX_NAME = 'endpoint-agent*';
 }
 
+export interface AlertResultList {
+  /* the alerts restricted by the page size */
+  alerts: AlertData[];
+  /* the total number of unique alerts in the index */
+  total: number;
+  /* the page size requested */
+  request_page_size: number;
+  /* the page index requested */
+  request_page_index: number;
```

Comment:
Is this the final design here? Why not a cursor based pagination approach? https://www.sitepoint.com/paginating-real-time-data-cursor-based-pagination/

Reply:
The elastic search has scroll and search after functionality. If you are not using collapse then this is an option for you. 

https://www.elastic.co/guide/en/elasticsearch/reference/6.8/search-request-collapse.html
```
collapse cannot be used in conjunction with scroll, rescore or search after.
```

Reply:
I don't believe we have any functionality in our mocks/design that would require `collapse`. I'll update to use a more sane pagination approach...

Reply:
@oatkiller I'm going to suggest that we save cursor-based pagination for a later PR if that's okay? There are some tricky things to figure out there, IMO ... `search_after` can certainly be used, but it becomes a little complicated when using that parameter and then changing the sort order... or paging backwards, for example.  We should probably talk about what we want this to look like, and how we want it to work.

Reply:
I agree with table this till when you need it. If you are receiving multiple event for a single entity (not sure what this is in your domain), and you want the latest then you need to aggregate and collapse is the least complicated way. You could also do an aggregation on your own using bucket et al'

Reply:
Created an issue to track cursor-based pagination. Resolving this.